### ISSUE TITLE
 file upload count suddenly mysteriously high for firebase deploy

### ISSUE THREAD

**Issue by VincentJ711** [firebase/firebase-tools#435]
firebase --version == 3.9.2

I deployed my app using firebase deploy --only hosting and it says:

i  deploying hosting
i  hosting: preparing ./ directory for upload...
✔  hosting: 1153 files uploaded successfully
i  starting release process (may take several minutes)...
✔  Deploy complete!

This is an absurd number of files uploaded. I've never seen this number so high in my deploy reports. Usually my file count is at ~100. Its strange because i'm ignoring all hidden directories and **/node_modules/** as directed in my firebase.json file, so i don't know where these files are coming from. This high number is also shown in the deployment history in my project console.

I dont know what extra files (if any) were deployed that are causing the upload file count to be so high. When i perform a wget on my projects url, only 107 files are downloaded. 107 is the correct # of files, not 1153. I don't know how to reproduce the problem as it has occurred out of nowhere for me.

Im just wondering if others have experienced this and thought i should mention the unexpected behavior i'm experiencing.

**Comment by kenpropel** [comment]
Looks like the ".git" directory is getting uploaded. Even tho it's listed in firebase.json "ignore" list. This isn't good…

**Comment by VincentJ711** [comment]
You're right @kenpropel .  I just removed .git from the directory and the deploy count dropped back down to normal. Then i replaced it and it went back up to over 1000.  My firebase json

```
{
  "hosting": {
    "public": "./",
    "ignore": [
      "firebase.json",
      "**/.*",
      "**/node_modules/**",
      "functions",
      "gulpfile.js"
    ],
    "rewrites": [{
      "source": "**",
      "destination": "/index.html"
    }]
  }
}
```
and my .git directory is in my public directory.

**Comment by shawky94** [comment]
any luck ?
